dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/CompanyUpload.kt

```diff
@@ -0,0 +1,60 @@
+package org.dataland.datalandbatchmanager.service
+
+import org.dataland.datalandbackend.openApiClient.api.CompanyDataControllerApi
+import org.dataland.datalandbackend.openApiClient.infrastructure.ClientException
+import org.dataland.datalandbackend.openApiClient.infrastructure.ServerException
+import org.dataland.datalandbackend.openApiClient.model.CompanyInformation
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import java.net.SocketTimeoutException
+
+/**
+ * Class for handling the upload of the company information retrieved from GLEIF to the Dataland backend
+ */
+@Service
+class CompanyUpload(
+    @Autowired private val companyDataControllerApi: CompanyDataControllerApi,
+) {
+    companion object {
+        const val MAX_RETRIES = 3
+        const val UNAUTHORIZED_CODE = 401
+        const val FORBIDDEN_CODE = 403
+    }
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Uploads a single Company to the dataland backend performing at most MAX_RETRIES retries.
+     * This function absorbs errors and logs them.
+     */
+    fun uploadSingleCompany(
+        companyInformation: CompanyInformation,
+    ) {
+        var counter = 0
+        while (counter < MAX_RETRIES) {
+            try {
+                logger.info(
+                    "Uploading company data for ${companyInformation.companyName} " +
+                        "(LEI: ${companyInformation.identifiers[0].identifierValue})",
+                )
+                companyDataControllerApi.postCompany(companyInformation)
+                break
+            } catch (exception: SocketTimeoutException) {
+                logger.error("Unexpected timeout occurred. Response was: ${exception.message}.")
+                counter++
+            } catch (exception: ClientException) {
+                logger.error("Unable to upload company data. Response was: ${exception.message}.")
+                if (exception.statusCode == UNAUTHORIZED_CODE || exception.statusCode == FORBIDDEN_CODE) {
+                    logger.error("Authorization failed, attempting to regenerate access token.")
```

Comment:
Wo im Code ist der "attempt to regenerate access token" ersichtlich?
Also wo steckt die "regenerate-token-if-invalid" Logik drin?

Reply:
@SiegfriedSobkowiak Wieso versuchen wie die Request nochmal, wenn die Authorization nicht geklappt hat?

Reply:
Das Problem hier waren in der Vergangenheit Tokens die abgelaufen waren. Die Tokens werden in den HttpClients geholt und in den ApiClients verwendet. Aber wenn ich das richtig sehe sollte das Problem in der aktuellen Implementierung nicht mehr vorkommen, weil vor jedem request der Token aktualisiert wird. @MarcTM01 kannst du das bestätigen? Falls ja könnte man den Block eigentlich entfernen.